Kuadrant/dns-operator
# Helm charts

This PR introduces a way to manage a DNS Operator [Helm Chart](https://helm.sh/). This is not meant to replace the way we are building and delivering our manifests (Kustomize, OLM) but to provide an alternative (complementary) way of delivering the operator.

This early implementation uses Kustomize to create the chart template, instead of creating and maintaining new ones with Helm, to later customize the Helm only settings with its `values.yaml`

**NOTES**

* This chart (and every other operator chart we package in the future: Limitador, Authorino, etc) bundles only the operator in the same fashion OLM does.
* The operator version matches the chart version
* This first release doesn't use helm templating, it provides only the manifests previously built by Kustomize.
* The sole repository for all our operator charts will be hosted in [Kuadrant/helm-charts](https://github.com/Kuadrant/helm-charts/).
* The "source of truth" (source code and releasing logic) will be provided in each operator repo, not in the helm charts repository above.
* The helm charts repository is [Kuadrant/helm-charts](https://github.com/Kuadrant/helm-charts/)
* Documentation will be provided once the repository and other operator charts are tested.
* The steps to try out the repository will be provided in the kuadrant helm charts repo.
* For the release of the package, it will be tested once merged with alpha pre releases.


**Verification Steps**

1. Create a local kind cluster
```sh
kind create cluster --name kuadrant-local
```

2. Deploy (install) the default (`latest`) DNS Operator chart
```sh
make helm-install
```

3. Verify the installed operator image:
```sh
kubectl get deployments/dns-operator-controller-manager -n dns-operator-system -o yaml | grep image:
```
it should return:
```sh
image: quay.io/kuadrant/dns-operator:latest
```

4. Build new manifests with specific version of the operator
```sh
make helm-build VERSION=0.4.0-alpha1
```

5. Upgrade the current installed operator to the freshly built one
```sh
make helm-upgrade
```

6. Check the installed operator image:
```sh
kubectl get deployments/dns-operator-controller-manager -n dns-operator-system -o yaml | grep image:
```

it should return:
```sh
image: quay.io/kuadrant/dns-operator:v0.4.0-alpha1
```

7. For uninstalling the chart:
```sh
make helm-uninstall
```
